### PR TITLE
Adjust formatting, typos, etc

### PR DESCRIPTION
**Pull request by iranzo**
Adjust formatting, typos, etc

### CODE REVIEW

File: docs/mkdocs/developer/index.md [hunk 1]
```diff
@@ -1,18 +1,21 @@
 # Design notes
 
 Kernel Module Management brings a new `Module` CRD.
-`Module` represents an out-of-tree kernel module that must be inserted into the cluster nodes’ kernel through a
-DriverContainer scheduled by a `DaemonSet`.
+
+`Module` represents an out-of-tree kernel module that must be inserted into the cluster nodes’ kernel through a `DriverContainer` scheduled by a `DaemonSet`.
```
Comment (ybettan) [inline docs/mkdocs/developer/index.md:5]:
```suggestion
`Module` represents an out-of-tree kernel module that must be inserted into the cluster nodes’ kernel through a `ModuleLoader` scheduled by a `DaemonSet`.
```

Comment (ybettan) [reply docs/mkdocs/developer/index.md]: Should we also replace all the occurrences of `DriverContainer` in the code to `ModuleLoader`?

Comment (iranzo) [reply docs/mkdocs/developer/index.md]: > Should we also replace all the occurrences of `DriverContainer` in the code to `ModuleLoader`?

Probably would make sense to do it, but my PR was attempting some minor rearranging, I saw that Quentin did a  bigger one with reorg of docs that might include it

File: docs/mkdocs/developer/index.md [hunk 2]
```diff
@@ -70,29 +73,31 @@ status:
...
 Virtually any Kubernetes distribution ships with its own kernel.
 It is thus challenging for a kernel module vendor to make DriverContainer images available for all kernels available
 in all Kubernetes distributions out there (or even for a subset of them).
 
-KMM supports in-cluster builds of DriverContainer images when those are not made available by the vendor.
-The in-cluster build system is able to build an image from any Git repository that contains a `Dockerfile`.
-It can then push said image to a user-provided repository, optionally using authentication provided in a Kubernetes secret.
+KMM supports in-cluster builds of DriverContainer images when those are not made available by the vendor. The in-cluster build system is able to build an image from any Git repository that contains a `Dockerfile`. It can then push said image to a user-provided repository, optionally using authentication provided in a Kubernetes secret.
+
+**Optional:** on upstream Kubernetes, we may want to deploy an in-cluster registry to host in-cluster built images.
```
Comment (ybettan) [inline docs/mkdocs/developer/index.md:83]: How about adding a an example using minikube to the doc? Here is a reference on how to do it:
* https://github.com/kubernetes-sigs/kernel-module-management/blob/ee344d6183cb16d649d7df592242cfff01c52610/.github/actions/create-minikube-cluster/action.yml#L30
* https://github.com/kubernetes-sigs/kernel-module-management/blob/ee344d6183cb16d649d7df592242cfff01c52610/.github/workflows/e2e.yaml#L43-L46

File: docs/mkdocs/developer/index.md [hunk 3]
```diff
@@ -70,29 +73,31 @@ status:
...
 
-**Optional:** on upstream Kubernetes, we may want to deploy an in-cluster registry to host in-cluster built images.  
 **On OCP**, the build mechanism would be BuildConfig (maybe Shipwright in the future) and we can leverage the
 integrated in-cluster registry.
```
Comment (ybettan) [inline docs/mkdocs/developer/index.md:86]: I would also add here that the internal registry isn't enabled by default on bare metal installations and needs to be enabled manually:  https://docs.openshift.com/container-platform/4.11/registry/configuring_registry_storage/configuring-registry-storage-baremetal.html

**Comment by iranzo** [comment]
Raised https://github.com/kubernetes-sigs/kernel-module-management/issues/245
-- 

Pablo Iranzo Gómez

GnuPG: 0x5BD8E1E4
Web: https://iranzo.io  ||  https://t.me/redken_bot

**Comment by ybettan** [comment]
Thanks for the issues @iranzo

**Comment by ybettan** [comment]
Will lgtm once squashed.

**Comment by iranzo** [comment]
done!
-- 

Pablo Iranzo Gómez

GnuPG: 0x5BD8E1E4
Web: https://iranzo.io  ||  https://t.me/redken_bot

File: docs/mkdocs/developer/index.md (hunk 2 above)
Comment (iranzo) [reply docs/mkdocs/developer/index.md:83]: Raised https://github.com/kubernetes-sigs/kernel-module-management/issues/243 for this

**Comment by ybettan** [comment]
@iranzo If the e2e job isn't triggered at some point (`in-cluster-build`) you will probably need to repush to try triggering it.

**Pull request merged**